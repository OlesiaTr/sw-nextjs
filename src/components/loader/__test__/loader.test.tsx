import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import Loader from '../loader';

describe('Loader', () => {
  it('renders the Rings spinner with correct styles', () => {
    const { asFragment } = render(<Loader />);

    const loaderElement = document.querySelector(
      '[data-testid="rings-loading"]',
    );
    expect(loaderElement).toBeInTheDocument();

    // Checks if the element has inline styles with correct values
    expect(loaderElement).toHaveStyle({
      position: 'fixed',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
