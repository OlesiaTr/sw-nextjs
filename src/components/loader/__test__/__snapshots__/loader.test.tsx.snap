// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders the Rings spinner with correct styles 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="rings-loading"
    class="sc-braxZu juNRvt"
    data-testid="rings-loading"
    role="progressbar"
    style="position: fixed; top: 50%; left: 50%; transform: translate(-50%, -50%);"
  >
    <svg
      data-testid="rings-svg"
      height="200"
      stroke="#0C0B0B"
      viewBox="0 0 45 45"
      width="200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
        stroke-width="2"
        transform="translate(1 1)"
      >
        <circle
          cx="22"
          cy="22"
          r="6"
          stroke-opacity="0"
        >
          <animate
            attributeName="r"
            begin="1.5s"
            calcMode="linear"
            dur="3s"
            repeatCount="indefinite"
            values="6;22"
          />
          <animate
            attributeName="stroke-opacity"
            begin="1.5s"
            calcMode="linear"
            dur="3s"
            repeatCount="indefinite"
            values="1;0"
          />
          <animate
            attributeName="stroke-width"
            begin="1.5s"
            calcMode="linear"
            dur="3s"
            repeatCount="indefinite"
            values="2;0"
          />
        </circle>
        <circle
          cx="22"
          cy="22"
          r="6"
          stroke-opacity="0"
        >
          <animate
            attributeName="r"
            begin="3s"
            calcMode="linear"
            dur="3s"
            repeatCount="indefinite"
            values="6;22"
          />
          <animate
            attributeName="strokeOpacity"
            begin="3s"
            calcMode="linear"
            dur="3s"
            repeatCount="indefinite"
            values="1;0"
          />
          <animate
            attributeName="strokeWidth"
            begin="3s"
            calcMode="linear"
            dur="3s"
            repeatCount="indefinite"
            values="2;0"
          />
        </circle>
        <circle
          cx="22"
          cy="22"
          r="8"
        >
          <animate
            attributeName="r"
            begin="0s"
            calcMode="linear"
            dur="1.5s"
            repeatCount="indefinite"
            values="6;1;2;3;4;5;6"
          />
        </circle>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
